convert_hf_to_gguf.py

```diff
@@ -5153,7 +5153,7 @@ def set_vocab(self):
         special_vocab._set_special_token("eos", tokenizer.get_added_vocab()["<|endoftext|>"])
         special_vocab._set_special_token("eot", tokenizer.get_added_vocab()["<|user|>"])
         special_vocab._set_special_token("unk", tokenizer.get_added_vocab()["<|endoftext|>"])
-        special_vocab._set_special_token("bos", tokenizer.get_added_vocab()["[gMASK]"])
+        special_vocab._set_special_token("bos", tokenizer.get_added_vocab()["<|endoftext|>"])
```

Comment:
Instead of changing BOS, I think what should be done is to set `add_add_bos(False)` to prevent tokenize function from adding BOS to the sentence